controllers/utils.go

```diff
@@ -64,6 +67,18 @@ func (r *RabbitmqClusterReconciler) statefulSet(ctx context.Context, rmq *rabbit
 	return sts, nil
 }
 
+func (r *RabbitmqClusterReconciler) statefulSetUID(ctx context.Context, rmq *rabbitmqv1beta1.RabbitmqCluster) (types.UID, error) {
+	uid := types.UID("")
+	if sts, err := r.statefulSet(ctx, rmq); err == nil {
+		if ref := metav1.GetControllerOf(sts); ref != nil {
+			if string(rmq.GetUID()) == string(ref.UID) {
+				return sts.UID, nil
+			}
+		}
+	}
+	return uid, fmt.Errorf("failed to get the uid of the statefulset owned by the current rabbitmqCluster")
+}
+
```

Comment:
Two suggestions: 1) I would add a line of comment to explain that  the statefulSet uid is returned successfully only when the controller reference uid matches the rabbitmqCluster uid. 2) I think errors of failing to get the statefulSet, failing to get controller reference,  and reference uid does not match rmq uid are all different failures. The function only returns one type of error at the moment, and it's unclear which step has failed. It's more verbose but the error msg is clearer:

```suggestion
func (r *RabbitmqClusterReconciler) statefulSetUID(ctx context.Context, rmq *rabbitmqv1beta1.RabbitmqCluster) (types.UID, error) {
	var err error
	var sts *appsv1.StatefulSet
	var ref *metav1.OwnerReference

	if sts, err = r.statefulSet(ctx, rmq); err != nil {
		return "", fmt.Errorf("failed to get statefulSet: %s", err.Error())
	}

	if ref = metav1.GetControllerOf(sts); ref == nil {
		return "", errors.New("failed to get controller reference for statefulSet")
	}

	if string(rmq.GetUID()) != string(ref.UID) {
		return "", errors.New("statefulSet not owner by current RabbitmqCluster")
	}

	return sts.UID, nil
}
```